src/input/input_python.rs

```diff
@@ -271,13 +271,6 @@ impl<'a> Input<'a> for PyAny {
     fn strict_int(&'a self) -> ValResult<EitherInt<'a>> {
         if PyInt::is_exact_type_of(self) {
             Ok(EitherInt::Py(self))
-        } else if PyInt::is_type_of(self) {
-            // bools are a subclass of int, so check for bool type in this specific case
-            if PyBool::is_exact_type_of(self) {
-                Err(ValError::new(ErrorType::IntType, self))
-            } else {
-                Ok(EitherInt::Py(self))
-            }
```

Comment:
This branch specifically feels like a bug. I think the int validator got around it because it immediately converts to a BigInt.

Reply:
I'm not sure I agree that this is a bug.

Naturally you'd expect subclasses of `int` to passing here.

Reply:
reverted / fixed